pallets/parachain-staking/src/lib.rs

```diff
@@ -1265,66 +1262,57 @@ pub mod pallet {
 			Ok(().into())
 		}
 		fn pay_stakers(next: RoundIndex) {
+			// payout is next - duration rounds ago => next - duration > 0 else return early
```

Comment:
One potential optimization I've been wanting to mention is to make a single storage write when one nominator has nominated multiple collators. I don't know how common this will be, but it's probably a strategy some people might consider...

Reply:
And, of course, the proper context for this is that writes are very expensive ;)

Actually, this might have some security implications.... :thinking:

Reply:
I expect a pretty complicated migration when I remove the multiple nominations from this pallet and constrain each `AccountId` to a single nomination. That change is going to remove a lot of complexity from the pallet implementation, but the migration will probably be complicated.

Reply:
Some approaches to the migration from simple to complex:
* All nominations are removed and need to be remade - might have security implications to change the staked amount so suddenly
* Each nominator only retains their top nomination, others are removed and need to be recreated with new accounts.
* All existing nominations are replaced with nominations from derived accounts (derived using pallet_utility) with sequential indices